platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ResolutionIssuesRegressionIntegrationTest.groovy

```diff
@@ -0,0 +1,576 @@
+
+            dependencies {
+              api(enforcedPlatform(project(":bom")))
+            }
+        """
+
+        file("main/bom/build.gradle.kts") << """
+            plugins {
+              id("java-platform")
+            }
+
+            group = "org.baz"
+
+            dependencies {
+              constraints {
+                api(project(":actions"))
+                api(project(":feature"))
+              }
+            }
+        """
+
+        file("main/feature/build.gradle.kts") << """
+            plugins {
+              id("java-library")
+              id("java-test-fixtures")
+            }
+
+            group = "org.baz"
+
+            dependencies {
+              api(enforcedPlatform(project(":bom")))
+            }
+        """
+
+        file("main/settings.gradle.kts") << """
+            include(":main")
+            include(":actions")
+            include(":external")
+            include(":bom")
+            include(":feature")
+        """
+
+        //endregion
+
+        file("settings.gradle.kts") << """
+            includeBuild(".")
+
+            dependencyResolutionManagement {
+              repositories {
+                maven(url = "${mavenRepo.uri.toString()}") {
+                  metadataSources {
+                    gradleMetadata()
+                  }
+                }
+              }
+            }
+
+            includeBuild("main") {
+              dependencySubstitution {
+                substitute(module("org.baz:main")).using(project(":main"))
+                substitute(module("org.baz:external")).using(project(":external"))
+              }
+            }
+
+            include(":uber-bom")
+            include(":base")
+        """
+
+        file("uber-bom/build.gradle.kts") << """
+            plugins {
+              id("java-platform")
+            }
+
+            group = "org.foo"
+
+            javaPlatform {
+              allowDependencies()
+            }
+
+            dependencies {
+              api(platform("org.baz:bom:1.2")) {
+                (this as ModuleDependency).doNotEndorseStrictVersions()
+              }
+            }
+        """
+
+        file("base/build.gradle.kts") << """
+            plugins {
+              id("java-library")
+            }
+
+            configurations.all {
+              resolutionStrategy {
+                preferProjectModules()
+              }
+            }
+
+            dependencies {
+                implementation(platform(project(":uber-bom")))
+                implementation(platform("org.junit:junit-bom:5.10.2"))
+                implementation("org.bar:client:1.1")
+                implementation("org.baz:external")
+                implementation("org.bar:client-testing:1.1")
+                implementation("org.junit.jupiter:junit-jupiter-params:5.11.3")
+            }
+        """
+
+        expect:
+        succeeds(":base:dependencies", "--configuration", "compileClasspath")
+    }
+
+    def "test"() {
```

Comment:
We should have a better name for this + rename the dependencies to something more generic (+ rebase)

Reply:
See my other comment on the simpler test that got fixed. I would consider using it instead of this one.

Reply:
I figured there was some value in keeping the complexity of this reproducer around, since it exercises some complex code paths of the engine that may not otherwise be handled elsewhere.

But, since it is not very targeted and since we have another reproducer that exercises this specific situation, I'm fine omitting it.